### PR TITLE
Speaker labelling 

### PR DESCRIPTION
**Pull request by afaisma**
### Summary
Added RecognizeUsingWebSocketsWithSpeakerLabelsExample. Note, that speaker labeling so far only works for En US narrowband, so it would make sense to create a new example class rather, than use the existing classes. 

The code changed according to German's suggestions

### CODE REVIEW

File: examples/java/com/ibm/watson/developer_cloud/speech_to_text/v1/RecognizeUsingWebSocketsWithSpeakerLabelsExample.java [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonObject;
+import com.ibm.watson.developer_cloud.http.HttpMediaType;
+import com.ibm.watson.developer_cloud.speech_to_text.v1.model.*;
+import com.ibm.watson.developer_cloud.speech_to_text.v1.websocket.BaseRecognizeCallback;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.util.*;
```
Comment (germanattanasio) [inline examples/java/com/ibm/watson/developer_cloud/speech_to_text/v1/RecognizeUsingWebSocketsWithSpeakerLabelsExample.java]: Don't use `*`

File: examples/java/com/ibm/watson/developer_cloud/speech_to_text/v1/RecognizeUsingWebSocketsWithSpeakerLabelsExample.java [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    void updateFrom(SpeechTimestamp speechTimestamp) {
+      word = speechTimestamp.getWord();
+    }
+    void updateFrom(SpeakerLabel speakerLabel) {
+      speaker = speakerLabel.getSpeaker();
+    }
+  }
+
+  static class Utterance {
+    private int speaker;
```
Comment (germanattanasio) [inline examples/java/com/ibm/watson/developer_cloud/speech_to_text/v1/RecognizeUsingWebSocketsWithSpeakerLabelsExample.java]: Use `Integer`

File: speech-to-text/src/main/java/com/ibm/watson/developer_cloud/speech_to_text/v1/model/SpeakerLabel.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import com.google.gson.annotations.SerializedName;
+
+/**
+ * Created by afaisman on 12/2/16.
+ */
+public class SpeakerLabel {
+
+  private Double from;
+  private Double to;
+  private int speaker;
```
Comment (germanattanasio) [inline speech-to-text/src/main/java/com/ibm/watson/developer_cloud/speech_to_text/v1/model/SpeakerLabel.java]: Use `Integer` not `int`

File: speech-to-text/src/main/java/com/ibm/watson/developer_cloud/speech_to_text/v1/model/SpeakerLabel.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+  public void setTo(Double to) {
+    this.to = to;
+  }
+
+  public int getSpeaker() {
+    return speaker;
+  }
+
+  public void setSpeaker(int speaker) {
```
Comment (germanattanasio) [inline speech-to-text/src/main/java/com/ibm/watson/developer_cloud/speech_to_text/v1/model/SpeakerLabel.java]: Integer

File: speech-to-text/src/main/java/com/ibm/watson/developer_cloud/speech_to_text/v1/model/SpeakerLabel.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    this.speaker = speaker;
+  }
+
+  public boolean isFinal() {
+    return _final;
+  }
+
+  public void setFinal(boolean ffinal) {
```
Comment (germanattanasio) [inline speech-to-text/src/main/java/com/ibm/watson/developer_cloud/speech_to_text/v1/model/SpeakerLabel.java]: `Boolean _final`

**Comment by germanattanasio** [comment]
@afaisma can you add some unit tests to the speech to text class? There are some examples that you can follow...

**Comment by germanattanasio** [comment]
Make sure you run `mvn clean install -Dmaven.test.skip=true` before committing code.

**Comment by afaisma** [comment]
Hi German,
 
Looks like the "check style" plugin disagrees with the name "_final". Would you mind if i change it back to "ffinal" as it was before?  
BTW, ffinal is used speech_to_text.v1.model.Transcript, where i took the idea from.
 
Thanks!
Alex
 
/Users/afaisman/dev/projects/adding-diarization-to-sdk/java-sdk/speech-to-text/src/main/java/com/ibm/watson/developer_cloud/speech_to_text/v1/model/RecognizeOptions.java:393:3: error: '{' at column 3 should be on the previous line./Users/afaisman/dev/projects/adding-diarization-to-sdk/java-sdk/speech-to-text/src/main/java/com/ibm/watson/developer_cloud/speech_to_text/v1/model/RecognizeOptions.java:393:3: error: '{' at column 3 should have line break after./Users/afaisman/dev/projects/adding-diarization-to-sdk/java-sdk/speech-to-text/src/main/java/com/ibm/watson/developer_cloud/speech_to_text/v1/model/SpeakerLabel.java:15:19: error: Name '_final' must match pattern '^[a-z][a-zA-Z0-9]*$'./Users/afaisman/dev/projects/adding-diarization-to-sdk/java-sdk/speech-to-text/src/main/java/com/ibm/watson/developer_cloud/speech_to_text/v1/model/SpeakerLabel.java:45:32: error: Name '_final' must match pattern '^[a-z][a-zA-Z0-9]*$'.
 
 
----- Original message -----From: German Attanasio <notifications@github.com>To: watson-developer-cloud/java-sdk <java-sdk@noreply.github.com>Cc: Alexander Faisman/Watson/IBM@IBMUS, Mention <mention@noreply.github.com>Subject: Re: [watson-developer-cloud/java-sdk] Speaker labelling (#486)Date: Sun, Dec 4, 2016 7:23 PM 
@germanattanasio requested changes on this pull request.
In examples/java/com/ibm/watson/developer_cloud/speech_to_text/v1/RecognizeUsingWebSocketsWithSpeakerLabelsExample.java:
> + * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the+ * specific language governing permissions and limitations under the License.+ */+package com.ibm.watson.developer_cloud.speech_to_text.v1;++import com.google.gson.Gson;+import com.google.gson.GsonBuilder;+import com.google.gson.JsonObject;+import com.ibm.watson.developer_cloud.http.HttpMediaType;+import com.ibm.watson.developer_cloud.speech_to_text.v1.model.*;+import com.ibm.watson.developer_cloud.speech_to_text.v1.websocket.BaseRecognizeCallback;++import java.io.FileInputStream;+import java.io.FileNotFoundException;+import java.util.*;
Don't use *
In examples/java/com/ibm/watson/developer_cloud/speech_to_text/v1/RecognizeUsingWebSocketsWithSpeakerLabelsExample.java:
> +    }+    RecoToken(SpeakerLabel speakerLabel) {+      startTime = speakerLabel.getFrom();+      endTime = speakerLabel.getTo();+      speaker = speakerLabel.getSpeaker();+    }+    void updateFrom(SpeechTimestamp speechTimestamp) {+      word = speechTimestamp.getWord();+    }+    void updateFrom(SpeakerLabel speakerLabel) {+      speaker = speakerLabel.getSpeaker();+    }+  }++  static class Utterance {+    private int speaker;
Use Integer
In speech-to-text/src/main/java/com/ibm/watson/developer_cloud/speech_to_text/v1/model/SpeakerLabel.java:
> @@ -0,0 +1,49 @@+package com.ibm.watson.developer_cloud.speech_to_text.v1.model;++import com.google.gson.annotations.SerializedName;++/**+ * Created by afaisman on 12/2/16.+ */+public class SpeakerLabel {++  private Double from;+  private Double to;+  private int speaker;
Use Integer not int
In speech-to-text/src/main/java/com/ibm/watson/developer_cloud/speech_to_text/v1/model/SpeakerLabel.java:
> +    this.from = from;+  }++  public Double getTo() {+    return to;+  }++  public void setTo(Double to) {+    this.to = to;+  }++  public int getSpeaker() {+    return speaker;+  }++  public void setSpeaker(int speaker) {
Integer
In speech-to-text/src/main/java/com/ibm/watson/developer_cloud/speech_to_text/v1/model/SpeakerLabel.java:
> +    this.to = to;+  }++  public int getSpeaker() {+    return speaker;+  }++  public void setSpeaker(int speaker) {+    this.speaker = speaker;+  }++  public boolean isFinal() {+    return _final;+  }++  public void setFinal(boolean ffinal) {
Boolean _final

**Comment by germanattanasio** [comment]
@afaisma makes sense. Sorry about that!

**Comment by afaisma** [comment]
No problem at all )
 
----- Original message -----From: German Attanasio <notifications@github.com>To: watson-developer-cloud/java-sdk <java-sdk@noreply.github.com>Cc: Alexander Faisman/Watson/IBM@IBMUS, Mention <mention@noreply.github.com>Subject: Re: [watson-developer-cloud/java-sdk] Speaker labelling (#486)Date: Mon, Dec 5, 2016 10:07 AM 
@afaisma makes sense. Sorry about that!

**Comment by afaisma** [comment]
Speaker Labeling (diarization) added

**Comment by afaisma** [comment]
Speaker Labeling (diarization) added

**Pull request closed without merging**